kong/runloop/plugins_iterator.lua

```diff
@@ -486,6 +492,28 @@ function PluginsIterator.new(version)
         end
 
       else
+        if version == "init" then
+          if not enabled_plugins[name] then
+            return nil, name .. " plugin is in use but not enabled"
+          end
```

Comment:
Is this check still performed elsewhere for declarative configuration?

Reply:
AFAIK this check did nothing on dbless as this point the declarative config was not loaded. Perhaps with reloads it did. But I think it is a good idea to add the check there too. Thanks for spotting it.

Reply:
@thibaultcha I fixed it and moved the check above. Now the plugin name is compared on each time the plugin's iterator is built, on all modes.